src/component/registry/types.ts

```diff
@@ -201,3 +203,26 @@ export type Font = {
    */
   isStrikeThrough?: boolean;
 };
+
+/**
+ * @category Element Style
+ */
+export type Fill = {
+  /**
+   * Possible values are all HTML color names or HEX codes, as well as special keywords such as ‘swimlane‘,
+   * ‘inherit’ to use the color code of a related cell.
+   */
+  color?: string | 'swimlane' | 'inherit';
+
+  opacity?: Opacity;
+};
+
+type CreateArrayWithLengthX<LENGTH extends number, RESULT extends number[] = []> = RESULT['length'] extends LENGTH ? RESULT : CreateArrayWithLengthX<LENGTH, [...RESULT, 1]>;
+type Range<MIN extends number, MAX extends number, RESULT extends number[] = CreateArrayWithLengthX<MIN>> = RESULT['length'] extends MAX
+  ? MIN | MAX
+  : Range<MIN | RESULT['length'], MAX, [...RESULT, 1]>;
+
+/**
+ * @category Element Style
+ */
+export type Opacity = Range<0, 100>;
```

Comment:
**issue (blocking):** 

As mentioned in https://stackoverflow.com/questions/39494689/is-it-possible-to-restrict-number-to-a-certain-range, it's not possible to create a dynamical union type with increment number before 4.5 version of TypeScript.

@tbouffard Do we change the minimal TypeScript version supported by `bpmn-visualization`, or the other solution is to list manually all the value of `Opacity` ?

Reply:
**Decision**: Finally, we choose with @tbouffard to remove the `Opacity` type, and use `number` type with validator.